### PR TITLE
feat: api integration new analytics

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Integrated api's for new analytics 
Screenshot 2024-10-01 at 10 29 42 AM
Screenshot 2024-10-01 at 10 29 30 AM
Screenshot 2024-10-01 at 10 29 24 AM

## How did you test it?
on changing the date range filter the data should update acordingly 

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [x] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/NewAnalytics/NewAnalyticsUtils.res [hunk 1]
```diff
@@ -34,3 +34,34 @@ let fillMissingDataPoints = (
...
+  ~excludeFilterValue as _: option<array<status>>=None,
+  ~applyFilterFor as _: option<array<status>>=None,
+  ~delta: option<bool>=None,
+  ~granularity: option<string>=None,
+) => {
+  let metrics = metrics->Array.map(v => (v: metrics :> string))
+  let filter = Dict.make()->JSON.Encode.object->Some
+
+  [
+    AnalyticsUtils.getFilterRequestBody(
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/NewAnalyticsUtils.res:57]: move this function inside NewAnalyticsUtils

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res [hunk 2]
```diff
@@ -93,92 +102,90 @@ let make = (
...
-        ~endDate="2024-08-27",
-        ~defaultValue={
-          count: 0,
-          amount: 0.0,
-          time_bucket: "",
-        }->Identity.genericTypeToJson,
-        ~timeKey="time_bucket",
-        ~granularity=granularity.value,
-      )->Identity.genericTypeToJson
+      //let url = getURL(~entityName=ANALYTICS_PAYMENTS, ~methodType=Post, ~id=Some(domain))
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res:120]: add screenState to loading

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res [hunk 3]
```diff
@@ -93,92 +102,90 @@ let make = (
...
+
+      let url = "https://integ-api.hyperswitch.io/analytics/v1/org/metrics/payments"
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res:122]: move this to API utils

File: src/screens/NewAnalytics/NewAnalyticsUtils.res (hunk 1 above)
Comment (sagarnaikjuspay) [reply src/screens/NewAnalytics/NewAnalyticsUtils.res:57]: its there in NewAnalyticsUtils utself

File: src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res [hunk 4]
```diff
@@ -1,13 +1,40 @@
...
   ->getArrayFromJson([])
   ->Array.flatMap(item => {
     item
     ->getDictFromJsonObject
     ->getArrayFromDict("queryData", [])
-    ->Array.map(item => item->getDictFromJsonObject->getString(key, ""))
+    ->Array.map(item => {
+      let value = item->getDictFromJsonObject->getString(key, "")
+
+      if value->LogicUtils.isNonEmptyString && key == "time_bucket" {
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/PaymentAnalytics/NewPaymentAnalyticsUtils.res:31]: LogicUtils is already defined above please remove this

File: src/screens/NewAnalytics/PaymentAnalytics/SuccessfulPaymentsDistribution/SuccessfulPaymentsDistributionUtils.res [hunk 5]
```diff
@@ -42,21 +45,34 @@ let getObjects: JSON.t => array<successfulPaymentsDistributionObject> = json =>
...
 
-let getCell = (obj, colType): Table.cell => {
+let getCell = (obj, colType: metrics): Table.cell => {
   switch colType {
-  | PaymentsSuccessRate => Text(obj.payments_success_rate->Int.toString)
-  | Connector => Text(obj.connector)
-  | PaymentMethod => Text(obj.payment_method)
+  | #payment_success_rate => Text(obj.payments_success_rate->Int.toString)
+  | #connector => Text(obj.connector)
+  | #payment_method | _ => Text(obj.payment_method)
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/PaymentAnalytics/SuccessfulPaymentsDistribution/SuccessfulPaymentsDistributionUtils.res:75]: do we need to handle `_` instead can we define all the metrics

File: src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res [hunk 6]
```diff
@@ -14,10 +15,17 @@ module TableModule = {
...
 
-    let paymentsProcessed =
-      data
-      ->LogicUtils.getArrayDataFromJson(tableItemToObjMapper)
-      ->Array.map(Nullable.make)
+    let paymentsProcessed = switch data->getArrayFromJson([])->Array.get(0) {
+    | Some(val) => {
+        let valueDict = val->getDictFromJsonObject
+        valueDict
+        ->getArrayFromDict("queryData", [])
```
Comment (JeevaRamu0104) [inline src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res:22]: in multiple places we are trying to extract the queryData.We need to make this generic

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by gitanjli525 — APPROVED** [review]

**Pull request merged**